will look at build
```
======================================================================
ERROR: testOtherUserReassignsIndexed (casexml.apps.phone.tests.test_sync_mode.MultiUserSyncTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 45, in inner
    return helper(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 32, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/tzmigration/test_utils.py", line 28, in call_with_settings
    fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 28, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 1145, in testOtherUserReassignsIndexed
    assert_user_has_case(self, self.user, case_id, restore_id=latest_sync_log.get_id)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 115, in assert_user_has_case
    return assert_user_has_cases(testcase, user, [case_id], return_single=True, **kwargs)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 121, in assert_user_has_cases
    should_have=True, line_by_line=False, version=V2, **kwargs)
  File "corehq/ex-submodules/casexml/apps/case/tests/util.py", line 160, in check_user_has_case
    payload_string = restore_config.get_payload().as_string()
  File "corehq/ex-submodules/casexml/apps/phone/restore.py", line 467, in get_payload
    partial_response = provider.get_response(self.restore_state)
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/providers.py", line 12, in get_response
    return get_case_payload(restore_state)
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py", line 22, in get_case_payload
    return sync_op.get_payload()
  File "corehq/ex-submodules/casexml/apps/phone/data_providers/case/clean_owners.py", line 126, in get_payload
    self.restore_state.current_sync_log.prune_dependent_cases()
  File "corehq/ex-submodules/casexml/apps/phone/models.py", line 704, in prune_dependent_cases
    self.prune_case(dependent_case_id)
  File "corehq/ex-submodules/casexml/apps/phone/models.py", line 595, in prune_case
    _remove_case(to_remove)
  File "corehq/ex-submodules/casexml/apps/phone/models.py", line 575, in _remove_case
    self.case_ids_on_phone.remove(to_remove)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/jsonobject/containers.py", line 182, in remove
    raise KeyError(wrapped)
KeyError: u'other_reassigns_index_parent'
----------------------------------------------------------------------
```

--- a/corehq/ex-submodules/casexml/apps/phone/models.py
+++ b/corehq/ex-submodules/casexml/apps/phone/models.py
@@ -700,8 +700,11 @@ class SimplifiedSyncLog(AbstractSyncLog):
         # this is done when migrating from old formats or during initial sync
         # to prune non-relevant dependencies
         for dependent_case_id in list(self.dependent_case_ids_on_phone):
-            # this will be a no-op if the case cannot be pruned due to dependencies
-            self.prune_case(dependent_case_id)
+            # need this additional check since the case might have already been pruned/remove
+            # as a result of pruning the child case
+            if dependent_case_id in self.dependent_case_ids_on_phone:
+                # this will be a no-op if the case cannot be pruned due to dependencies
+                self.prune_case(dependent_case_id)
 
     @classmethod
     def from_other_format(cls, other_sync_log):

--- a/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
+++ b/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
@@ -1152,6 +1152,10 @@ class MultiUserSyncTest(SyncBaseTest):
         payload = generate_restore_payload(self.project, self.user, latest_sync_log.get_id, version=V2)
         self.assertTrue("something new" in payload)
         self.assertTrue("hi!" in payload)
+        # also check that the latest sync log knows those cases are no longer relevant to the phone
+        log = synclog_from_restore_payload(payload)
+        self.assertFalse(log.phone_is_holding_case(case_id))
+        self.assertFalse(log.phone_is_holding_case(parent_id))
         
         # change the parent again from the second user
         other_parent_update = CaseBlock(

--- a/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
+++ b/corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py
@@ -1141,7 +1141,10 @@ class MultiUserSyncTest(SyncBaseTest):
         
         # original user syncs again
         latest_sync_log = SyncLog.last_for_user(self.user.user_id)
-        # both cases should sync to original user with updated ownership / edits
+
+        # at this point both cases are assigned to the other user so the original user
+        # should not have them. however, the first sync should send them down (with new ownership)
+        # so that they can be purged.
         assert_user_has_case(self, self.user, case_id, restore_id=latest_sync_log.get_id)
         assert_user_has_case(self, self.user, parent_id, restore_id=latest_sync_log.get_id)
 
@@ -1159,7 +1162,6 @@ class MultiUserSyncTest(SyncBaseTest):
             update={"other_greeting": "something different"}, 
             version=V2).as_xml()
         self._postFakeWithSyncToken(other_parent_update, other_sync_log.get_id)
-        
         
         # original user syncs again
         latest_sync_log = SyncLog.last_for_user(self.user.user_id)

